### PR TITLE
Feature/apvs 344/add dwp check task

### PR DESCRIPTION
**Pull request by stevenalexander**
- added dwp check task
- insert DWP check after submitting claim

## Checklist

- [ ] Unit tests
- [ ] Code coverage checked
- [ ] Coding standards
- [ ] Error Handling
- [ ] Security/performance considered?
- [ ] Deployment changes considered?
- [ ] README updated

### CODE REVIEW

File: app/services/benefit-checker/call-dwp-benefit-checker-soap-service.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  return `<?xml version="1.0" encoding="utf-8"?>
+<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <soapenv:Body>
+    <benefitCheckerRequest xmlns="https://lsc.gov.uk/benefitchecker/service/1.0/API_1.0_Check">
+      <lscServiceName>${config.DWP_BENEFIT_CHECKER_LSCSERVICENAME}</lscServiceName>
+      <clientOrgId>${config.DWP_BENEFIT_CHECKER_CLIENTORGID}</clientOrgId>
+      <clientUserId>${config.DWP_BENEFIT_CHECKER_CLIENTUSERID}</clientUserId>
+      <clientReference>${config.DWP_BENEFIT_CHECKER_CLIENTREFERENCE}</clientReference>
+      <nino>${data.nino}</nino>
+      <surname>${data.surname}</surname>
```
Comment (tmrowe) [inline app/services/benefit-checker/call-dwp-benefit-checker-soap-service.js]: Handling for surnames with special characters.

File: app/services/get-worker-for-task.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ var sendRejectedClaimNotification = require('./workers/send-rejected-claim-notif
 var sendRequestInformationClaimNotification = require('./workers/send-request-information-claim-notification')
 var sendFirstTimeClaimNotification = require('./workers/send-first-time-claim-notification')
 var completeFirstTimeClaim = require('./workers/complete-first-time-claim')
+var dwpCheck = require('./workers/dwp-check')
 
 // ALL WORKERS SHOULD HAVE A METHOD `execute(task)` that returns a Promise
 module.exports = function (taskType) {
```
Comment (tmrowe) [inline app/services/get-worker-for-task.js]: Maybe switch here?

File: test/test-helper.js [hunk 3]
```diff
@@ -123,6 +123,10 @@ module.exports.insertClaimEligibilityData = function (schema, reference) {
       delete data.ClaimExpenses[1].ClaimExpenseId
       data.ClaimExpenses[0].ClaimId = newClaimId
       data.ClaimExpenses[1].ClaimId = newClaimId
+      if (isIntSchema) {
+        delete delete data.ClaimExpenses[0].IsEnabled
```
Comment (tmrowe) [inline test/test-helper.js]: Remove one delete.

File: test/unit/services/test-get-worker-for-task.js [hunk 4]
```diff
@@ -46,4 +48,10 @@ describe('services/getWorkerForTask', function () {
     expect(worker.name).to.be.equal('completeFirstTimeClaim')
     done()
   })
+
+  it('should return dwp-check', function (done) {
+    var worker = getWorkerForTask(tasksEnum.DWP_CHECK)
+    expect(worker.name).to.be.equal('dwpCheck')
+    done()
```
Comment (tmrowe) [inline test/unit/services/test-get-worker-for-task.js]: Remove done.

File: config.js [hunk 5]
```diff
@@ -20,5 +20,11 @@ module.exports = {
   NOTIFY_ACCEPTED_CLAIM_EMAIL_TEMPLATE_ID: process.env.APVS_NOTIFY_ACCEPTED_CLAIM_EMAIL_TEMPLATE_ID,
   NOTIFY_REJECTED_CLAIM_EMAIL_TEMPLATE_ID: process.env.APVS_NOTIFY_REJECTED_CLAIM_EMAIL_TEMPLATE_ID,
   NOTIFY_REQUEST_INFORMATION_CLAIM_EMAIL_TEMPLATE_ID: process.env.APVS_NOTIFY_REQUEST_INFORMATION_CLAIM_EMAIL_TEMPLATE_ID,
-  NOTIFY_FIRST_TIME_CLAIM_EMAIL_TEMPLATE_ID: process.env.APVS_NOTIFY_FIRST_TIME_CLAIM_EMAIL_TEMPLATE_ID
+  NOTIFY_FIRST_TIME_CLAIM_EMAIL_TEMPLATE_ID: process.env.APVS_NOTIFY_FIRST_TIME_CLAIM_EMAIL_TEMPLATE_ID,
+  // DWP Benefit Checker
```
Comment (tmrowe) [inline config.js]: Line breaks

File: config.js [hunk 6]
```diff
@@ -20,5 +20,11 @@ module.exports = {
...
+  DWP_BENEFIT_CHECKER_URL: process.env.APVS_DWP_BENEFIT_CHECKER_URL,
+  DWP_BENEFIT_CHECKER_LSCSERVICENAME: process.env.APVS_DWP_BENEFIT_CHECKER_LSCSERVICENAME,
+  DWP_BENEFIT_CHECKER_CLIENTORGID: process.env.APVS_DWP_BENEFIT_CHECKER_CLIENTORGID,
+  DWP_BENEFIT_CHECKER_CLIENTUSERID: process.env.APVS_DWP_BENEFIT_CHECKER_CLIENTUSERID,
+  DWP_BENEFIT_CHECKER_CLIENTREFERENCE: process.env.APVS_DWP_BENEFIT_CHECKER_CLIENTREFERENCE
 }
```
Comment (tmrowe) [inline config.js]: Add env variable to disable the checker.

File: app/services/benefit-checker/call-dwp-benefit-checker-soap-service.js (hunk 1 above)
Comment (stevenalexander) [inline app/services/benefit-checker/call-dwp-benefit-checker-soap-service.js]: sent email to DWP to check on this, so far the service still responds 200 with apostrophe in the name but would need valid data to test

**Pull request merged**